### PR TITLE
fix(tomcat-10.1): Build packages for supported JDK versions

### PR DESCRIPTION
**Pull request by EyeCantCU**
Allows us to bundle the correct JDK version with the package and run tests for each supported JDK version at the package level

### CODE REVIEW

File: tomcat-10.1.yaml [hunk 1]
```diff
@@ -35,78 +40,77 @@ pipeline:
...
+subpackages:
+  - range: openjdk-versions
+    name: ${{package.name}}-${{range.value}}
+    dependencies:
+      runtime:
+        - ${{range.value}}-default-jvm
+        - tomcat-native
+    pipeline:
+      - runs: |
+          export JAVA_HOME="/usr/lib/jvm/java-17-openjdk"
```
Comment (pnasrat) [inline tomcat-10.1.yaml:52]: java-${{range.key}}

Comment (EyeCantCU) [reply tomcat-10.1.yaml:52]: Tomcat builds with OpenJDK 17, but supports 11-21 at runtime. We benefit from tests against all JVM versions supported at runtime, but we do end up building duplicate packages

Comment (pnasrat) [reply tomcat-10.1.yaml:56]: ah ok can you add a comment to that effect!

Comment (EyeCantCU) [reply tomcat-10.1.yaml:56]: Yes, I can definitely do that :slightly_smiling_face:

**Review by pnasrat — APPROVED** [review]
LGTM

**Pull request merged**